### PR TITLE
Feature/switch program

### LINKED ISSUE #5929: Need a better way to switch default program
There are multiple `program` when we training a neural network using `fluid` API. They are:

1. The training program(s). There could be many training programs if we training a complex model, like GAN, RL, etc.
2. The testing program(s). Testing programs used to evaluate the accuracy or cost on test data set while training. We use an independent program while testing because there are some operators using a different equation when testing, such as `dropout`.

We need a better way to easily switch default program for layers in our API.

This issue will be done in two steps:

1. remove `g_main_program`/`g_startup_program`. Use `default_main_program` and `default_startup_program` instead.
2. Add API `switch_program` to change default programs.

### PR DESCRIPTION
**Pull request by reyoung**
Second step of #5929.
Fix  #5929.

Please review #5930 first.

### CODE REVIEW

File: python/paddle/v2/fluid/framework.py [hunk 1]
```diff
@@ -659,8 +661,83 @@ def __init__(self, block, shape, dtype, **kwargs):
...
+    Args:
+        program(Program): The new main program
+
+    Returns:
+        Program: The previous main program
+    """
+    global _main_program_
+    prev_program = _main_program_
+    _main_program_ = program
+    return prev_program
```
Comment (QiJune) [inline python/paddle/v2/fluid/framework.py]: Here, why return `prev_program`, not `_main_program_`? We are switch to a new program, not previous one

File: python/paddle/v2/fluid/framework.py [hunk 2]
```diff
@@ -659,8 +661,83 @@ def __init__(self, block, shape, dtype, **kwargs):
...
+    """
     return _startup_program_
 
 
 def default_main_program():
+    """
+    Get default main program. The main program is used for training or testing.
+    
+    Returns:
+        Program: main program
```
Comment (QiJune) [inline python/paddle/v2/fluid/framework.py]: Why we need this `default_main_program()` function, not use `_main_program_` directly?
